src/components/NavigationMailbox.vue

```diff
@@ -444,6 +481,40 @@ export default {
 		onCloseMoveModal() {
 			this.showMoveModal = false
 		},
+		onDragStart({ accountId }) {
+			if (accountId !== this.mailbox.accountId) {
+				return
+			}
+			this.$store.commit('expandAccount', accountId)
+			this.showSubMailboxes = true
+		},
+		onDragEnd({ accountId }) {
+			if (accountId !== this.mailbox.accountId) {
+				return
+			}
+			this.showSubMailboxes = false
+		},
+		onEnvelopesMoved({ mailboxId, movedEnvelopes }) {
+			if (this.mailbox.databaseId !== mailboxId) {
+				return
+			}
+			this.$store.dispatch('syncEnvelopes', { mailboxId })
```

Comment:
I think we don't need/want that, do we? The destination mailbox is always a different one than the current mailbox. And we don't care about keeping another mailbox updated (on the front-end).

@GretaD found out that this action can run into https://github.com/nextcloud/mail/blob/f37124c3def79072bbcd2a32d71eded92533117a/lib/Controller/MailboxesController.php#L171. That is, a sync request is sent for a mailbox that had never been sync'ed before. In this case the js code is actually supposed to do a full initialization sync request. So, that is of course out of scope for the drag and drop feature, but this dispatched action could potentially carry out a very expensive operation because the initial sync isn't happening fast for larger mailboxes.

Hence I'd say we remove the explicit sync of the destination mailbox for now. When the user navigates there, the background sync will update the mailbox anyway. https://github.com/nextcloud/mail/issues/3951 is a bit of a missing piece here, but again, off topic :wink: 

@brueckner does that make sense or am I missing something?

Reply:
Hmmm, I see the point. However, before I implemented triggering the sync it was pretty irritating to me to move a mail to its new destination and not seeing it. Sometimes it would take quite some time until the background sync hit. Until then – from the perspective of an unaware user – the mail is just... gone. I don't know about you, but I would be pretty angry if a mail app (seemingly) made my mails disappear 👀 . Also: wouldn't the background-sync essentially do the same expensive operation as well, just... delayed? ;)

_(Disclaimer: I didn't look at the background sync yet)_ Couldn't we reset the timer of a background sync based on a move-operation's triggered sync or something along those lines? 

OR... yes! Hear me out... What if we dropped the sync-trigger and _marked_ every mailbox that was the target of a recent move-operation instead (something like `outdated`). If a user clicks on a mailbox that is marked `outdated`, as sync is triggered immediately. That way, the user will be greeted by a loading spinner and glad to find that all their mails have been moved successfully. On the other hand, all `outdated` mailboxes the user does **not** navigate to will not trigger a sync (until the regular background sync hits). 

That way I can move a thousand mails to 20 different mailboxes while only ever triggering one sync once I navigate to one of the 20 `outdated` mailboxes. In addition to that, the background sync will remove the outdated flag, thus preventing an unnecessary sync.

👯

Reply:
Are you describing https://github.com/nextcloud/mail/issues/3951? :thinking:

Reply:
Ha! Hell yea! :)

So in terms of what's described in #3951 we could just disable the sync on any move-target-mailbox in the store without persisting that setting?

Reply:
Well, wait. It would just have to be disabled for like 30 seconds, right? If the user has bg sync enabled, moves a mail into the folder and **doesn't** open the mailbox right away afterwards, the background sync should still to its' thing on regular schedule.

Reply:
While #3951 is very specific about non-background sync'ed mailboxes we can discuss always firing off a sync based on a *last sync* attribute on the mailbox, as you suggested.

> If the user has bg sync enabled, moves a mail into the folder and **doesn't** open the mailbox right away afterwards, the background sync should still to its' thing on regular schedule.

I think that depends on how much we want to keep mailboxes updated in the front-end code if they aren't open right now. I'd say if the front-end never fetched a list of messages, there is no point in spending energy on updating this mailbox. The user might have 30 mailboxes subscribed, this could be a lot of work for the server.

What matters most is that the currently open mailbox is relatively up to date. Same applies to when you switch to another mailbox and the messages are loaded from the database cache: there should be a sync to ensure we always show the latest data.

Reply:
Btw we have the same problems/questions for drafts (they should show up if you switch mailbox) and also sent messages.

I'm all in for improving that. But can we keep those two things separate?

Reply:
Yea sure – so should I just remove the sync-trigger? We'd just have to make sure to have a solution for it when this gets released, right?

Reply:
> We'd just have to make sure to have a solution for it when this gets released, right?

Absolutely. We can discuss this the next few days (maybe in a call) and then have a solid fix for the next release :)

Reply:
Sound good to me! I'll remove the trigger :)

Reply:
Removed, rebased on master and squashed. ✅